toolkit/themes/shared/design-system/config.js

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* eslint-env node */
+
+const StyleDictionary = require("style-dictionary");
+const { fileHeader } = StyleDictionary.formatHelpers;
+
+/**
+ * CSS formatter that converts tokens that have a light and dark value
+ * to a light-dark() function. If we try to use the built-in CSS formatter,
+ * we will lose the "darkValue" or any other property that isn't "value".
+ */
+function lightDarkFormatter(args) {
+  let dictionary = args.dictionary;
+  let file = args.file;
+  const selector = args.options.selector ? args.options.selector : `:root`;
+  const indentation = args.options.indentation ? args.options.indentation: 2;
+  let modifiedDictionary = dictionary.allTokens.map(token => {
+    let value = JSON.stringify(token.value);
+    // the `dictionary` object now has `usesReference()` and
+    // `getReferences()` methods. `usesReference()` will return true if
+    // the value has a reference in it. `getReferences()` will return
+    // an array of references to the whole tokens so that you can access their
+    // names or any other attributes.
+    if (dictionary.usesReference(token.original.value) && token.original.darkValue) {
+      // Note: make sure to use `token.original.value` because
+      // `token.value` is already resolved at this point.
+      const lightRefs = dictionary.getReferences(token.original.value);
+      const darkRefs = dictionary.getReferences(token.original.darkValue);
+      for (let i = 0; i < darkRefs.length; i++) {
```

Comment:
It seems like we'll always have 1 item in this array, if we don't then this is just hiding a bug I think

This got me wondering *why* it's even an array though. Looks like we could do something like:

```json
"brand": {
  "value": "",
  "lightDarkValue": ["{color.gray.100}", "{color.gray.05}"]
}
```

Which is now making me wonder if we could transform directly on the `value` key?

```json
{
  "text": {
    "color": {
      "value": "CanvasText"
    },
    "brand": {
      "value": ["{color.gray.100}", "{color.gray.05}"]
    }
  }
}
```

Reply:
I do wonder if it'll make things easier for iOS / Android to have the `dark` values spelled out explicitly since they break things up into themes...though I guess that's still many steps down the line and having to grab the second value in an array isn't complicated 🤷🏻

Reply:
I think we can use the current W3C design tokens structure and use the composite form (https://design-tokens.github.io/community-group/format/#composite-design-token) to handle this case. For example:
```json
{
"foo": {
    "value": {
      "light": "white",
      "dark": "black",
      "forcedColors": "CanvasText",
    }
  },
  "bar": {
    "value": {
      "light": "{color.gray.100}",
      "dark": "{color.gray.05}"
    }
  }
}
```
and then use a formatter to convert the values accordingly.